pkg/kubeapiserver/options/authentication.go

```diff
@@ -269,6 +269,25 @@ func (o *BuiltInAuthenticationOptions) Validate() []error {
 
 	if o.RequestHeader != nil {
 		allErrors = append(allErrors, o.RequestHeader.Validate()...)
+		if o.ClientCert != nil {
+			// Enforce 'requestheader-allowed-names' check when 'requestheader-client-ca-file'
+			// and 'client-ca-file' are the same and not empty, to prevent unauthorized privilege escalation.
+			clientCA := strings.TrimSpace(o.ClientCert.ClientCA)
+			requestHeaderClientCA := strings.TrimSpace(o.RequestHeader.ClientCAFile)
+			if len(clientCA) != 0 && clientCA == requestHeaderClientCA {
+				numValidNames := 0
+				for _, allowedName := range o.RequestHeader.AllowedNames {
+					if len(strings.TrimSpace(allowedName)) != 0 {
```

Comment:
no need to TrimSpace, that's already guarded by checkForWhiteSpaceOnly, can just check `len(o.RequestHeader.AllowedNames) == 0`